kube/src/config/file_config.rs

```diff
@@ -200,34 +202,27 @@ impl AuthInfo {
                         }
 
                         if let Some(field) = provider.config.get("token-key") {
-                             // incredbly naive JSONPath impl
-                            let splitted = field
-                                .trim_matches(|c| c == '"' || c == '{' || c == '}')
-                                .split("." )
-                                .filter(|a| a.len() > 0);
-                            let mut current : serde_json::Value = serde_json::from_slice(&output.stdout)?;
-                            for item in splitted {
-                                if let serde_json::Value::Object(mut map) = current {
-                                    if let Some(v) = map.remove(item) {
-                                        current = v;
+                            let pure_path = field
+                                .trim_matches(|c| c == '"' || c == '{' || c == '}');
+                            let json_output : serde_json::Value = serde_json::from_slice(&output.stdout)?;
+                            match jsonpath_select(&json_output, &format!("${}", pure_path)) {
```

Comment:
is this a quirk of jsonpath lib that all selectors need to start with a `$`?

Reply:
no, I think it is jsonpath-wide, that it must start with `$`, [at least other evaluators also fail if I try without](https://freestrings.github.io/jsonpath/#%24..*) (but I haven't checked the spec tbh).

Reply:
turns out it's just kubernetes treating them as optional:

> The $ operator is optional since the expression always starts from the root object by default.
https://kubernetes.io/docs/reference/kubectl/jsonpath/

so this is fine :-)